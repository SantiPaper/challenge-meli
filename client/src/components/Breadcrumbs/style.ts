import styled from "styled-components";

export const StyledBreadcrumbs = styled.ul`
  padding: var(--spacing-m) 0;
  display: flex;

  .item {
    font-size: var(--font-size-200);
    list-style: none;
    color: var(--gray-400);

    &:last-child {
      color: var(--gray-600);
      font-weight: 600;
    }

    &:not(:last-child)::after {
      content: ">";
      margin: 1.5rem;
    }
  }

  .link {
    text-decoration: none;
    color: currentColor;
  }

  li:not(:last-child)::after {
    content: ">";
    margin: 1.5rem;
  }
`;
