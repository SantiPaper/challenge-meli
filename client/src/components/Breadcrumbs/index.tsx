import { Categories } from "../../types/items";
import { StyledBreadcrumbs } from "./style";

type Props = {
  categories: Categories;
};
export const Breadcrumbs = ({ categories }: Props) => {
  return (
    <StyledBreadcrumbs>
      <ul className="breadcrumbs">
        {categories.map((catergory) => (
          <li key={catergory} className="breadcrumbs__item">
            <a href="#" className="breadcrumbs__item__link">
              {catergory}
            </a>
          </li>
        ))}
      </ul>
    </StyledBreadcrumbs>
  );
};
