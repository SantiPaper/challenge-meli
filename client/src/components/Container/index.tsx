import { FC, PropsWithChildren } from "react";
import { StyledContainer } from "./style";

export const Container: FC<PropsWithChildren> = ({ children }) => {
  return <StyledContainer>{children}</StyledContainer>;
};
