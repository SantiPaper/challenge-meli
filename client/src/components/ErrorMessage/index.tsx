import { StyledErrorMessage } from "./style";

type Props = {
  error: string | null;
};

export const ErrorMessage = ({ error }: Props) => {
  return (
    <StyledErrorMessage>
      <h2 className="text">{error}</h2>
    </StyledErrorMessage>
  );
};
