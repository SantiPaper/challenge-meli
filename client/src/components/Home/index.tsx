import { StyledHome } from "./style";

export const Home = () => {
  return (
    <StyledHome>
      <h1 className="hero">
        Aun no realizaste tu b√∫squeda, te alentamos a que lo hagas!
      </h1>
    </StyledHome>
  );
};
