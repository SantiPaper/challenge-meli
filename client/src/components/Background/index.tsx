import { FC, PropsWithChildren } from "react";
import { Container } from "../Container";
import { StyledBackground } from "./style";

export const Background: FC<PropsWithChildren> = ({ children }) => {
  return (
    <Container>
      <StyledBackground>{children}</StyledBackground>
    </Container>
  );
};
