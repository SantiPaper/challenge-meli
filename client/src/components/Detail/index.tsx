import { useLocation } from "react-router-dom";
import { useFetch } from "../../hooks/useFetch";
import { useProductsContext } from "../../hooks/useProductsContext";
import { Breadcrumbs } from "../Breadcrumbs";
import { Loading } from "../Loading";
import { Background } from "../Background";
import { StyledDetail } from "./style";

export const Detail = () => {
  const location = useLocation();

  const { state, handleFormat } = useProductsContext();

  const product = state.product.item;

  const categories = state.product.categories;

  const { loading } = useFetch({
    status: "/",
    payload: location.pathname.split("/")[2],
  });

  return (
    <StyledDetail>
      {loading ? (
        <Loading />
      ) : (
        <>
          <Breadcrumbs categories={categories} />
          <Background>
            <div className="detail__info">
              <img
                className="detail__info__img"
                src={product.picture}
                alt={product.title}
                width={680}
                height={680}
              />

              <div className="detail__info__aside">
                <p className="detail__info__aside__condition">
                  {product.condition === "new" ? "Nuevo" : "Usado"}
                </p>
                <h1 className="detail__info__aside__title">{product.title}</h1>
                <p className="detail__info__aside__price">
                  {handleFormat(product.price.amount)}
                </p>
                <button className="detail__info__aside__button">Comprar</button>
              </div>
            </div>
            <div className="detail__description">
              <h3 className="detail__description__title">
                Descripci√≥n del producto
              </h3>
              <p className="detail__description__text">{product.description}</p>
            </div>
          </Background>
        </>
      )}
    </StyledDetail>
  );
};
