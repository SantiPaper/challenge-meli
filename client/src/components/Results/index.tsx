import { useSearchParams } from "react-router-dom";
import { useProductsContext } from "../../hooks/useProductsContext";
import { useFetch } from "../../hooks/useFetch";
import { Background } from "../Background";
import { ItemProduct } from "../ItemProduct";
import { Breadcrumbs } from "../Breadcrumbs";
import { Loading } from "../Loading";
import { Container } from "../Container";
import { ErrorMessage } from "../ErrorMessage";

export const Results = () => {
  const { state } = useProductsContext();
  const [params] = useSearchParams();

  const { loading, error } = useFetch({
    status: "?q=",
    payload: params.get("search"),
  });

  return (
    <section>
      {loading ? (
        <Loading />
      ) : (
        <>
          {state.searchResults.categories && (
            <Breadcrumbs categories={state.searchResults.categories} />
          )}
          {state.searchResults.items ? (
            <Background>
              {state.searchResults.items.map((prod) => {
                const number = new Intl.NumberFormat("es-AR", {
                  style: "currency",
                  currency: "ARS",
                  minimumFractionDigits: 0,
                  maximumFractionDigits: 0,
                }).format(prod.price.amount);

                return (
                  <ItemProduct
                    key={prod.id}
                    id={prod.id}
                    title={prod.title}
                    price={number}
                    shipping={prod.free_shipping}
                    img={prod.picture}
                    condition={prod.condition}
                  />
                );
              })}
            </Background>
          ) : (
            <Container>
              <ErrorMessage error={error} />
            </Container>
          )}
        </>
      )}
    </section>
  );
};
