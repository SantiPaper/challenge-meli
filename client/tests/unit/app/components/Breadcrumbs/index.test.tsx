import { render, screen } from "@testing-library/react";
import { Breadcrumbs } from "../../../../../src/components/Breadcrumbs";
import { mockCategories } from "../Results/mocks";
import userEvent from "@testing-library/user-event";

beforeEach(() => {});

describe("Breadcrumbs", () => {
  test("Renderiza breadcrumbs", () => {
    const { container } = render(<Breadcrumbs categories={mockCategories} />);
    expect(container).toMatchSnapshot();
  });
  test("Se puede navegar entre breadcrumbs", async () => {
    render(<Breadcrumbs categories={mockCategories} />);
    const breadCrumbList = screen.getAllByRole("link");
    const mockFn = vi.fn();
    breadCrumbList[0].onclick = mockFn;
    await userEvent.click(breadCrumbList[0]);
    expect(mockFn).toHaveBeenCalledTimes(1);
  });
});
