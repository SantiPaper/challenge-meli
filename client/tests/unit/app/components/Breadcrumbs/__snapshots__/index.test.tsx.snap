// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Breadcrumbs > Renderiza breadcrumbs 1`] = `
<div>
  <nav
    class="sc-beySPh gQqSYr"
  >
    <ul
      class="breadcrumbs"
    >
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__item__link"
          href="#"
        >
          Category 1
        </a>
      </li>
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__item__link"
          href="#"
        >
          Category 2
        </a>
      </li>
      <li
        class="breadcrumbs__item"
      >
        <a
          class="breadcrumbs__item__link"
          href="#"
        >
          Category 3
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
